game-app/game-core/src/main/java/games/strategy/triplea/ui/history/HistoryLog.java

```diff
@@ -515,7 +515,7 @@ public void printProductionSummary(final GameData data) {
     final Collection<GamePlayer> players;
     final Resource pus;
     try (GameData.Unlocker ignored = data.acquireReadLock()) {
-      pus = data.getResourceList().getResource(Constants.PUS);
+      pus = data.getResourceList().getResource(Constants.PUS).orElse(null);
```

Comment:
@BBlaza Same, either the method assumes the value must be available, then new method `getResourceOrThrow?` or a default value should be used, e.g., `orElse(0)`.